lobster/tools/coda/coda.py

```diff
@@ -0,0 +1,391 @@
+#!/usr/bin/env python3
+#
+# lobster_coda - Extract C/C++ tracing tags from commands for LOBSTER
+# Copyright (C) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program. If not, see
+# <https://www.gnu.org/licenses/>.
+
+import json
+import sys
+import argparse
+import os.path
+from copy import copy
+from enum import Enum
+
+from lobster.items import Tracing_Tag, Activity
+from lobster.location import File_Reference
+from lobster.io import lobster_write
+from lobster.tools.coda.parser.constants import LOBSTER_GENERATOR
+from lobster.tools.coda.parser.requirements_parser import ParserForRequirements
+
+OUTPUT  = "output"
+MARKERS = "markers"
+KIND    = "kind"
+
+NAMESPACE_CPP = "cpp"
+LANGUAGE_CPP = "C/C++"
+FRAMEWORK_CPP_TEST = "cpp_test"
+KIND_FUNCTION = "Function"
+CB_PREFIX = "CB-#"
+MISSING = "Missing"
+
+
+class RequirementTypes(Enum):
+    REQS = '@requirement'
+    REQ_BY = '@requiredby'
+    DEFECT = '@defect'
+
+
+SUPPORTED_REQUIREMENTS = [
+    RequirementTypes.REQS.value,
+    RequirementTypes.REQ_BY.value,
+    RequirementTypes.DEFECT.value
+]
+
+map_test_type_to_key_name = {
+    RequirementTypes.REQS.value: 'requirements',
+    RequirementTypes.REQ_BY.value: 'required_by',
+    RequirementTypes.DEFECT.value: 'defect_tracking_ids',
+}
+
+
+def parse_config_file(file_name: str) -> dict:
```

Comment:
please add another feature to read the codebeamer ULR, in order to replace the `CODEBEAMER_URL` constant in `constants.py`